apps/src/feedback.js

```diff
@@ -370,6 +375,48 @@ FeedbackUtils.prototype.displayFeedback = function (options, requiredBlocks,
   }
 };
 
+FeedbackUtils.prototype.calculateStageProgress = function (hintsUsed) {
+  const stage = this.studioApp_.reduxStore.getState().progress.stages[0];
+  const scriptName = stage.script_name;
+  const levels = stage.levels;
+  const progress = ClientState.allLevelsProgress();
+  const oldFinishedHints = authoredHintUtils.getOldFinishedHints();
+
+  let numPassed = 0,
+    numPerfect = 0,
+    numZeroHints = 0,
+    numOneHint = 0;
+  for (let i = 0; i < levels.length; i++) {
+    const levelProgress = Math.max.apply(Math, levels[i].ids.map(id => progress[scriptName][id]));
```

Comment:
Does `clientState.js` already have a function for returning the variant with the most progress?  If not, this logic should maybe live there.